### DISCUSSION TITLE
Questions about execution limit

### DISCUSSION THREAD

**Discussion by computerwhizguy** [sebastienros/jint#1781] [Q&A]
I use a product, that includes JINT, and they have set an execution limit of 100,000.  In order to optimize usage:

1. Is there a way to tell your current execution line count?  I understand that checking it will burn at least a line, but if you are trying to avoid an exception it might be helpful.
2. What is the best way to tell how it is counting. I have tried using good programming practices within the JS, however, given the limit, changing some strategies may pay dividends by conserving lines, at the cost of maintainability and readability.  For example, I was thinking of changing an if with braces and a single line then clause like this

if (foo=="bar") {
bleh = "baa";
}

if (foo=="bar") bleh="baa";

But I think thru testing I determined that it is counted at 2 lines either way.

However, an opportunity I think I found was to combine multiple variable declaration lines into a single line.

**Comment by lahma** [comment]
> 1. Is there a way to tell your current execution line count?  I understand that checking it will burn at least a line, but if you are trying to avoid an exception it might be helpful.

There's no way currently for JS code to know the running value OOTB at the moment.

> 2. What is the best way to tell how it is counting. I have tried using good programming practices within the JS, however, given the limit, changing some strategies may pay dividends by conserving lines, at the cost of maintainability and readability.  For example, I was thinking of changing an if with braces and a single line then clause like this
> 
> if (foo=="bar") { bleh = "baa"; }
> 
> if (foo=="bar") bleh="baa";
> 
> But I think thru testing I determined that it is counted at 2 lines either way.

It's statement limit so braces don't matter here.

> However, an opportunity I think I found was to combine multiple variable declaration lines into a single line.

Variable declaration statement can indeed have multiple items to initialize, which is counted as one statement.  Maybe you should ask whether they could give you less restrictive setup if you truly need more statements to be run?
